controllers/admin/AdminImportController.php

```diff
@@ -1260,161 +1805,180 @@ public function categoryImport($offset = false, $limit = false, &$crossStepsVari
-                        );
+                        $this->errors[] = $this->trans('%1$s (ID: %2$s) cannot be saved', array(
+                            $categoryToCreate->name[$idLang],
+                            !empty($categoryToCreate->id) ? $categoryToCreate->id : 'null',
+                        ), 'Admin.Advparameters.Notification');
                     }
-                    if ($field_error !== true || isset($lang_field_error) && $lang_field_error !== true) {
-                        $this->errors[] = ($field_error !== true ? $field_error : '').(isset($lang_field_error) && $lang_field_error !== true ? $lang_field_error : '').
-                            Db::getInstance()->getMsgError();
+                    if ($fieldError !== true || isset($langFieldError) && $langFieldError !== true) {
+                        $this->errors[] = ($fieldError !== true ? $fieldError : '')
+                            . (isset($langFieldError) && $langFieldError !== true ? $langFieldError : '')
+                            . Db::getInstance()->getMsgError();
                     }
                 }
             }
         }
-        if (isset($category->link_rewrite) && !empty($category->link_rewrite[$default_language_id])) {
-            $valid_link = Validate::isLinkRewrite($category->link_rewrite[$default_language_id]);
+        if (isset($category->link_rewrite) && !empty($category->link_rewrite[$defaultLanguageId])) {
+            $validLink = Validate::isLinkRewrite($category->link_rewrite[$defaultLanguageId]);
         } else {
-            $valid_link = false;
+            $validLink = false;
         }
 
-        if (!$shop_is_feature_active) {
+        if (!$shopIsFeatureActive) {
             $category->id_shop_default = 1;
         } else {
             $category->id_shop_default = (int)Context::getContext()->shop->id;
         }
 
-        $bak = $category->link_rewrite[$default_language_id];
-        if ((isset($category->link_rewrite) && empty($category->link_rewrite[$default_language_id])) || !$valid_link) {
-            $category->link_rewrite = Tools::link_rewrite($category->name[$default_language_id]);
+        $bak = $category->link_rewrite[$defaultLanguageId];
+        if ((isset($category->link_rewrite) && empty($category->link_rewrite[$defaultLanguageId])) || !$validLink) {
+            $category->link_rewrite = Tools::link_rewrite($category->name[$defaultLanguageId]);
             if ($category->link_rewrite == '') {
                 $category->link_rewrite = 'friendly-url-autogeneration-failed';
-                $this->warnings[] = sprintf($this->trans('URL rewriting failed to auto-generate a friendly URL for: %s', array(), 'Admin.Advparameters.Notification'), $category->name[$default_language_id]);
+                $this->warnings[] = $this->trans('URL rewriting failed to auto-generate a friendly URL for: %s',
+                    array($category->name[$defaultLanguageId]),
```

Comment:
Same comment about translation placeholder. Please use a variable name surrounded by `%%` an add it in the second parameter.